lib/Controller/UserApiController.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace OCA\Text\Controller;
+
+use \OCP\AppFramework\ApiController;
+use \OCP\IRequest;
+use \OCP\IUserManager;
+
+class UserApiController extends ApiController {
+
+    protected IUserManager $userManager;
+
+    public function __construct($appName, IRequest $request, IUserManager $userManager) {
+        parent::__construct($appName, $request);
+
+        $this->userManager = $userManager;
+    }
+
+    /**
+     * @param string $filter
+     */
+    public function index(string $filter) {
+		$result = [];
+
+        $users = $this->userManager->search($filter, 5);
```

Comment:
A dedicated endpoint makes sense, since then we can later on sort the active users first. However for search ing users would be better to use https://github.com/nextcloud/server/blob/215aef3cbdc1963be1bb6bca5218ee0a4b7f1665/lib/public/Collaboration/Collaborators/ISearch.php#L40 as this will respect the privacy settings that may be in place on the instance to only display users that are allowed to be mentioned.

shareTypes would be `[0]` to only search for users.

See also https://github.com/nextcloud/server/blob/215aef3cbdc1963be1bb6bca5218ee0a4b7f1665/apps/files_sharing/lib/Controller/ShareesAPIController.php#L228 how this is used on the API for file sharing

Reply:
To inject `UserBubble` we need to make sure that the `UserBubble` will be handled as a VueJs component.

Reply:
Okay, I see what you mean. But there is a concern tough: how can we access the user's id because we need it?

Reply:
The format is a bit different but it should return you a list of users, from the following screenshot you should be able to see the format - llabel is the display name and `value['shareWith']` the user id:

<img width="1552" alt="Screenshot 2022-08-11 at 18 58 06" src="https://user-images.githubusercontent.com/3404133/184191199-3f576d71-ea05-4328-895c-30df7fdaa479.png">